vendor/github.com/Microsoft/opengcs/service/gcsutils/remotefs/remotefs.go

```diff
@@ -287,90 +288,115 @@ func Mkfifo(in io.Reader, out io.Writer, args []string) error {
 //  - args[1] = flag in base 10
 //  - args[2] = permission mode in octal (like 0755)
 func OpenFile(in io.Reader, out io.Writer, args []string) (err error) {
+	logrus.Debugf("OpenFile: %v", args)
+
 	defer func() {
 		if err != nil {
+			logrus.Errorf("OpenFile: return is non-nil, so writing cmdFailed back: %v", err)
 			// error code will be serialized by the caller, so don't write it here
 			WriteFileHeader(out, &FileHeader{Cmd: CmdFailed}, nil)
 		}
 	}()
 
 	if len(args) < 3 {
+		logrus.Errorf("OpenFile: Not enough parameters")
 		return ErrInvalid
 	}
 
 	flag, err := strconv.ParseInt(args[1], 10, 32)
 	if err != nil {
+		logrus.Errorf("OpenFile: Invalid flag: %v", err)
```

Comment:
I do agree with @thaJeztah, a good rule with errors is : handle them (do something about it, return it, …) or log them but not do both. Looking at those, it would be better to use `github.com/pkg/errors` and wrap them. Something like : 

```
return errors.Wrap(err, "Invalide flag")
```